### PR TITLE
PostGIS enhancements

### PR DESCRIPTION
**Pull request by ian-r-rose**
1. Allows for null geometries (it previously failed in unhelpful ways upon encountering those)
2. Add types for some missing geometry types, namely multilinestrings and multipoints.

I still need to add some tests, but wanted to put this out for comment.

### CODE REVIEW

**Comment by ian-r-rose** [comment]
@cpcloud I've added a few tests exercising the ibis APIs, but I have not done any testing against the actual PostGIS backend, which will likely take more time, and will involve changes to the test data. If that's acceptable to you for the time-being, this should be good from my perspective. Otherwise I can add more tests next week.

The CI failures *seem* to be unrelated, and afflicting master as well.

File: ibis/expr/datatypes.py [hunk 1]
```diff
@@ -1637,9 +1720,18 @@ def can_cast_variadic(
 
 # geo spatial data type
 # cast between same type, used to cast from/to geometry and geography
-@castable.register(Array, (Point, LineString, Polygon, MultiPolygon))
-@castable.register((Point, LineString, Polygon, MultiPolygon), Geometry)
-@castable.register((Point, LineString, Polygon, MultiPolygon), Geography)
+@castable.register(
```
Comment (xmnlab) [inline ibis/expr/datatypes.py]: would be good to create a new variable to group all geo types? 
maybe something like:

```
GEO_TYPES = (Point, LineString, Polygon, MultiLineString, MultiPoint, MultiPolygon)

@castable.register(Array, GEO_TYPES)
@castable.register(GEO_TYPES, Geometry)
@castable.register(GEO_TYPES, Geography)
def can_cast_geospatial(source, target, **kwargs):
    return True

Comment (ian-r-rose) [reply ibis/expr/datatypes.py]: Done!

Comment (xmnlab) [reply ibis/expr/datatypes.py]: great! thanks!

**Comment by xmnlab** [comment]
@ian-r-rose related to `mock tests` (https://github.com/ibis-project/ibis/blob/master/ibis/expr/tests/mocks.py#L423), I think it makes more sense now changing the compiler and dialect from omniscidb to postgres. what do you think?

**Comment by ian-r-rose** [comment]
@xmnlab I think that makes sense -- it might take me a day or two to get to it.

**Comment by xmnlab** [comment]
@ian-r-rose, great! I can work on that, I will open a PR for that and you can pick the commit from there.

**Comment by xmnlab** [comment]
@ian-r-rose related to smoke test I am working on this PR https://github.com/ibis-project/ibis/pull/1928

**Comment by cpcloud** [comment]
@ian-r-rose Can you rebase and force push to pick up some changes that should make your builds pass?

**Comment by ian-r-rose** [comment]
Rebased

**Comment by ian-r-rose** [comment]
In case there's any interest, I put together a binder with a local PostGIS server running to demo some of this work:
[Binder](https://mybinder.org/v2/gh/ian-r-rose/binder-postgis/master?urlpath=lab%2Ftree%2Fpostgis-ibis-geopandas.ipynb)

**Comment by ian-r-rose** [comment]
Thanks for the review @cpcloud, and enjoy your break!

And thanks for the help with testing @xmnlab.

**Pull request merged**